src/query/ast/src/ast/format/ast/mod.rs

```diff
@@ -0,0 +1,131 @@
+// Copyright 2022 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+mod dcl;
+mod ddl;
+mod dml;
+mod expr;
+mod query;
+mod show;
+
+use common_exception::Result;
+use dcl::*;
+use ddl::*;
+use dml::*;
+use query::*;
+use show::*;
+
+use crate::ast::ExplainKind;
+use crate::ast::FormatTreeNode;
+use crate::ast::Statement;
+
+pub fn format_statement(stmt: Statement) -> Result<String> {
+    format_stmt(stmt).format_pretty()
+}
+
+fn format_stmt(stmt: Statement) -> FormatTreeNode<String> {
```

Comment:
can we use a common visitor to apply the `format` ?

Reply:
ok